File: dojo20100519 - pedra_papel_tesoura - javascript/pedra_papel_tesoura.js
Jogador = function(nome, jogada){
    this.nome = nome;
    this.jogada = jogada;
}
function ganha(jogada1,jogada2){
    if jogada1 =='tesoura'
        return true
}

function ganhador(jogador1, jogador2){
    if (jogador1.jogada.ganha(jogador2.jogada))
        return jogador1.nome
    else
        return jogador2.nome
}

function jokenpo(jogadores) {

    this.todosIguais = function(){

        if(jogadores[0].jogada == jogadores[1].jogada){
            return true;
        }

    }


    if(jogadores.length == 1)
        return [jogadores[0].nome];
     else
        if (this.todosIguais()){
            return [jogadores[0].nome, jogadores[1].nome];
        }
        if (jogadores[0].jogada == "tesoura" &&
         jogadores[1].jogada == "papel") {
            return [jogadores[0].nome];
        }
        else if (jogadores[0].jogada== "papel" &&
        jogadores[1].jogada== "pedra"){
            return [jogadores[0].nome];
        }
        else{
            return [jogadores[1].nome];
        }
}

File: 2011/20110323 - pedra papel tesoura - ruby/pedra_papel_tesoura.rb
class Jogada 
    def initialize(jogada, quem_ganha_de)
        @jogada = jogada
        @quem_ganha_de = quem_ganha_de
    end

    def empata_com?(jogada)
        @jogada == jogada.jogada
    end
    
    def ganha_de?(jogada)
        @quem_ganha_de[jogada] == @jogada
    end
end


class Jokenpo
    @@quem_ganha_de = { 
            "papel" => "tesoura",
            "tesoura" => "pedra",
            "pedra" => "papel"
    }

    def initialize(jogada1, jogada2)
        @jogada1 = Jogada.new(jogada1, @@quem_ganha_de)
        @jogada2 = Jogada.new(jogada2, @@quem_ganha_de)
    end

    def resultado
        if @jogada1.empata_com? @jogada2
            "empate"
        elsif @jogada1.ganha_de? @jogada2
            @jogada1.to_s
        else
            @jogada1.to_s
        end
    end
end


